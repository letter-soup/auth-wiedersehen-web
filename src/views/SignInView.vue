<script setup lang="ts">
import { Button } from '@/components/ui/button'
import { Input } from '@/components/ui/input'
import { Label } from '@/components/ui/label'

const loginHandler = () => {
  console.log('debug:SignInView.vue:loginHandler; ')
}
</script>

<template>
  <div class="w-full lg:grid lg:min-h-[600px] lg:grid-cols-2 xl:min-h-[800px]">
    <div class="flex items-center justify-center py-12">
      <div class="mx-auto grid w-[350px] gap-6">
        <div class="grid gap-2 text-center">
          <h1 class="text-3xl font-bold">
            {{$t('sign-in:header')}}
          </h1>
          <p class="text-balance text-muted-foreground">
            {{$t('sign-in:description')}}
          </p>
        </div>
        <div class="grid gap-4">
          <div class="grid gap-2">
            <Label for="email">{{$t('sign-in:email')}}</Label>
            <Input
              id="email"
              type="email"
              placeholder="m@example.com"
              required
            />
          </div>
          <div class="grid gap-2">
            <div class="flex items-center">
              <Label for="password">{{$t('sign-in:password')}}</Label>
              <a
                href="/reset-password"
                class="ml-auto inline-block text-sm underline"
              >
                {{$t('sign-in:password-reset')}}
              </a>
            </div>
            <Input id="password" type="password" required />
          </div>
          <Button type="submit" class="w-full" @click="loginHandler">
            {{$t('sign-in:cta')}}
          </Button>
        </div>
        <div class="mt-4 text-center text-sm">
          {{$t('sign-in:no-account')}}
          <a href="/sign-up" class="underline">
            {{$t('sign-in:sign-up')}}
          </a>
        </div>
      </div>
    </div>
    <div class="hidden bg-muted lg:block">
      <img
        src="https://cdn.pixabay.com/photo/2024/06/01/12/28/monday-8802107_1280.jpg"
        alt="image"
        class="h-dvh w-full object-cover dark:brightness-[0.2] dark:grayscale"
      >
    </div>
  </div>
</template>
