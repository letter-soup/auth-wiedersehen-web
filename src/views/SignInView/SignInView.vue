<script setup lang="ts">
import { Button } from '@/components/ui/button'
import { Input } from '@/components/ui/input'
import { Label } from '@/components/ui/label'
import { Separator } from '@/components/ui/separator'
import { toast } from '@/components/ui/toast'

const signInHandler = () => {
  toast({
    title: 'You successfully signed in!',
  })
}
</script>

<template>
  <div class="w-full h-full flex items-center justify-center py-12">
    <div class="mx-auto grid w-[350px] gap-6">
      <div class="grid gap-2 text-center">
        <h1 class="text-3xl font-bold">
          {{ $t('sign-in:header') }}
        </h1>
        <p class="text-balance text-muted-foreground">
          {{ $t('sign-in:description') }}
        </p>
      </div>
      <div class="grid gap-4">
        <div class="grid gap-2">
          <Label for="email">{{ $t('sign-in:email') }}</Label>
          <Input id="email" type="email" placeholder="m@example.com" required />
        </div>
        <div class="grid gap-2">
          <div class="flex items-center">
            <Label for="password">{{ $t('sign-in:password') }}</Label>
            <RouterLink
              to="/reset-password"
              class="ml-auto inline-block text-sm underline text-muted-foreground hover:text-primary"
            >
              {{ $t('sign-in:password-reset') }}
            </RouterLink>
          </div>
          <Input id="password" type="password" required />
        </div>
        <Button type="submit" class="w-full" @click="signInHandler">
          {{ $t('sign-in:cta') }}
        </Button>
      </div>

      <Separator class="mt-2" :label="$t('sign-in:no-account')" orientation="horizontal" />
      <Button variant="link">
        <RouterLink to="/sign-up">
          {{ $t('sign-in:sign-up') }}
        </RouterLink>
      </Button>
    </div>
  </div>
</template>
