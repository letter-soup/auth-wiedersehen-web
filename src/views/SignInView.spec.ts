import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import SignInView from '@/views/SignInView.vue'

describe('SignInView (snapshot)', () => {
  it('renders properly', async () => {
    const wrapper = mount(SignInView)

    expect(wrapper).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
