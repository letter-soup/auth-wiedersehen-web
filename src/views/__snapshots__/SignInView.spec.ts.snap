// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignInView (snapshot) > renders properly 1`] = `
"<div class="w-full lg:grid lg:min-h-[600px] lg:grid-cols-2 xl:min-h-[800px]">
  <div class="flex items-center justify-center py-12">
    <div class="mx-auto grid w-[350px] gap-6">
      <div class="grid gap-2 text-center">
        <h1 class="text-3xl font-bold">Sign In</h1>
        <p class="text-balance text-muted-foreground">Enter your email below to login to your account</p>
      </div>
      <div class="grid gap-4">
        <div class="grid gap-2"><label for="email" class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70">Email</label><input class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50" id="email" type="email" placeholder="m@example.com" required=""></div>
        <div class="grid gap-2">
          <div class="flex items-center"><label for="password" class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70">Password</label><a href="/reset-password" class="ml-auto inline-block text-sm underline">Forgot your password?</a></div><input class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50" id="password" type="password" required="">
        </div><button class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2 w-full" type="submit">Sign In</button>
      </div>
      <div class="mt-4 text-center text-sm">Don't have an account? <a href="/sign-up" class="underline">Sign up</a></div>
    </div>
  </div>
  <div class="hidden bg-muted lg:block"><img src="https://cdn.pixabay.com/photo/2024/06/01/12/28/monday-8802107_1280.jpg" alt="image" class="h-dvh w-full object-cover dark:brightness-[0.2] dark:grayscale"></div>
</div>"
`;
